src/main/java/net/wimpi/modbus/net/ModbusTCPListener.java

```diff
@@ -160,13 +162,20 @@ public void stop() {
     @Override
     public void run() {
         try {
+            Channel inheritedChannel = System.inheritedChannel();
+            if (inheritedChannel instanceof ServerSocketChannel) {
+                ServerSocketChannel serverSocketChannel = (ServerSocketChannel) inheritedChannel;
+                m_ServerSocket = serverSocketChannel.socket();
```

Comment:
A bit later, on the line 179, there's a log statement that logs the port:
`logger.debug("Listenening to {} (Port {})", m_ServerSocket.toString(), m_Port);`

If the socket was created via this inherited channel then that line might be lying.
Therefore, assign the actual port value to that field:

```suggestion
                m_ServerSocket = serverSocketChannel.socket();
                m_Port = m_ServerSocket.getLocalPort();
```